lib/crack/json.rb

```diff
@@ -4,20 +4,32 @@
 # THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 require 'strscan'
+require 'psych'
```

Comment:
I don't believe this change is in scope with ths pr

Reply:
if I remember correctly, I had to add this to fix a failing test with one of the Ruby versions. I'm going to check it again and come back with the results

Reply:
Without 644b0f7 / #74 , the CI fails for Ruby 3.1, 3.2, 3.3 and JRuby equivalents, a sample run could be found here: https://github.com/kiskoza/crack/actions/runs/7622992428/job/20762118748

How should we resolve this?
1. Keep the commit fixing the code
2. Keep the Ruby versions within the CI matrix, but merge them as failing
3. Remove the newer Ruby versions and add them along the fix in a new PR

I would prefer 1 or 2 and wouldn't recommend doing 3, but happy to go the path you find the best

Reply:
I think simplest is to merge this with passing ci (comment out or allow failures) then in fix pr add the failing matrix options

Reply:
I've updated the CI matrix to allow failures for recent Ruby versions, it should be fine now. I'd happily help as being a maintainer as well.